zeebe/util/src/main/java/io/camunda/zeebe/util/SemanticVersion.java

```diff
@@ -21,6 +22,12 @@
 public record SemanticVersion(
     int major, int minor, int patch, String preRelease, String buildMetadata)
     implements Comparable<SemanticVersion> {
+
+  // The values in the map are all Optional<SemanticVersion> to avoid allocating a new optional
+  // every time. Optional.empty is forbidden.
+  private static final ConcurrentHashMap<String, SemanticVersion> CACHE =
```

Comment:
❓ 💭 Hm, I have the feeling we ran into issues when we use static here. Especially with running out of tests and CI.

Could we make it an instance field, and create the SemanticVersion object only once in the exporters? 🤔

Reply:
Thinking about it more, I guess it is not really an issue, as it is just about creating the semantic version, and then if we use different versions, we simply return this (and cache it).